File: bindings/java-jni/src/main/java/org/maplibre/nativejni/error/NativeErrorException.java
package org.maplibre.nativejni.error;

/** Thrown when a native Maplibre error or C++ exception crosses the C ABI as a status. */
public final class NativeErrorException extends MaplibreException {
  public NativeErrorException(int nativeStatusCode, String diagnostic) {
    super(MaplibreStatus.NATIVE_ERROR, nativeStatusCode, diagnostic);
  }
}

File: bindings/dotnet/src/Maplibre.Native/Error/NativeErrorException.cs
namespace Maplibre.Native.Error;

/// <summary>Exception for native MapLibre errors converted to C status.</summary>
public sealed class NativeErrorException : MaplibreException
{
    public NativeErrorException(
        MaplibreStatus status,
        int? rawStatus,
        string diagnostic,
        Exception? innerException = null
    )
        : base(status, rawStatus, diagnostic, innerException) { }
}
